tensorflow/contrib/tensorrt/resources/trt_allocator.cc

```diff
@@ -37,8 +37,22 @@ void TRTCudaAllocator::free(void* memory) { cudaFree(memory); }
 
 void* TRTDeviceAllocator::allocate(uint64_t size, uint64_t alignment,
                                    uint32_t flags) {
+  // WAR for allocator alignment requirement
+  alignment = 512;
   assert((alignment & (alignment - 1)) == 0);  // zero or a power of 2.
-  void* mem = allocator_->AllocateRaw(alignment, size);
+  void* mem = allocator_->AllocateRaw(alignment, size + alignment);
+
+  CHECK(mem_pool.count(mem) == 0);
```

Comment:
Instead of calling count, it would be more efficient to check the return value of insert to avoid doing two lookups, i.e.
```
CHECK(mem_pool.insert(mem).second);
```